OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseTabsDashboardFragment.kt

```diff
@@ -0,0 +1,886 @@
+                        showShiftDateSnackBar(false)
+                    }
+                }
+            }
+        })
+    }
+
+    override fun onResume() {
+        super.onResume()
+        EventBus.getDefault().register(this)
+        courseDateViewModel.fetchCourseDates(courseData.courseId, true)
+    }
+
+    override fun onPause() {
+        super.onPause()
+        EventBus.getDefault().unregister(this)
+        // TODO: block of code can be removed once `fetchCourseById` retrofit call replaced with MVVM approach.
+        fullscreenLoader?.closeTimer()
+    }
+
+    /**
+     * Method to handle the tab-selection of the ViewPager based on screen name [Screen]
+     * which may be sent through a deep link.
+     *
+     * @param bundle arguments
+     */
+    private fun handleTabSelection(bundle: Bundle?) {
+        bundle?.let {
+            @ScreenDef val screenName = bundle.getString(Router.EXTRA_SCREEN_NAME)
+            if (screenName != null && !bundle.getBoolean(Router.EXTRA_SCREEN_SELECTED, false)) {
+                for (i in courseTabItems.indices) {
+                    val item = courseTabItems[i]
+                    if (shouldSelectFragment(item, screenName)) {
+                        binding.pager.currentItem = i
+                        break
+                    }
+                }
+                // Setting `EXTRA_SCREEN_SELECTED` to true, so that upon recreation of the fragment the tab defined in
+                // the deep link is not auto-selected again.
+                bundle.putBoolean(Router.EXTRA_SCREEN_SELECTED, true)
+            }
+        }
+    }
+
+    /**
+     * Determines if a tab fragment needs to be selected based on screen name.
+     *
+     * @param item       [FragmentItemModel] assigned to a tab.
+     * @param screenName screen name param coming from [org.edx.mobile.deeplink.DeepLinkManager]
+     * @return `true` if the specified tab fragment needs to be selected, `false` otherwise
+     */
+    private fun shouldSelectFragment(
+        item: FragmentItemModel,
+        @ScreenDef screenName: String
+    ): Boolean {
+        return (screenName == Screen.COURSE_VIDEOS && item.title == getString(R.string.videos_title)) ||
+                (screenName == Screen.COURSE_DISCUSSION && item.title == getString(R.string.discussion_title)) ||
+                (screenName == Screen.DISCUSSION_POST && item.title == getString(R.string.discussion_title)) ||
+                (screenName == Screen.DISCUSSION_TOPIC && item.title == getString(R.string.discussion_title)) ||
+                (screenName == Screen.COURSE_DATES && item.title == getString(R.string.label_dates)) ||
+                (screenName == Screen.COURSE_HANDOUT && item.title == getString(R.string.handouts_title)) ||
+                (screenName == Screen.COURSE_ANNOUNCEMENT && item.title == getString(R.string.announcement_title))
+    }
+
+    private fun setViewPager() {
+        binding.pager.visibility = View.VISIBLE
```

Comment:
better to combine all the property setters of `binding.pager` or use binding.pager.apply{}
also please prefer to use `setVisibility` utility method in kotlin classes

Reply:
We can write this
```
private fun setViewPager() {
        setupToolbar(true)

        binding.pager.apply {
            setVisibility(true)
            enforceSingleScrollDirection(this)
            adapter = FragmentItemPagerAdapter(requireActivity(), courseTabItems)
            registerOnPageChangeCallback(object : OnPageChangeCallback() {
                override fun onPageSelected(position: Int) {
                    super.onPageSelected(position)
                    val item = courseTabItems[position]
                    if (item.listener != null) {
                        item.listener?.onFragmentSelected()
                    }
                }
            })
        }

        binding.toolbar.tabs.addOnTabSelectedListener(object : OnTabSelectedListener {
            override fun onTabSelected(tab: TabLayout.Tab) {
                binding.pager.currentItem = tab.position
            }

            override fun onTabUnselected(tab: TabLayout.Tab) {}
            override fun onTabReselected(tab: TabLayout.Tab) {}
        })

        TabLayoutMediator(
            binding.toolbar.tabs,
            binding.pager
        ) { tab: TabLayout.Tab, position: Int ->
            tab.text = courseTabItems[position].title
        }.attach()

        if (courseTabItems.size - 1 > 1) {
            binding.pager.offscreenPageLimit = courseTabItems.size - 1
        }
    }